src/test/java/pro/trevor/tankgame/rule/council/SpawnLavaTest.java

```diff
@@ -55,8 +69,9 @@ public void testPlayerCanHaveDeadTank() {
         GenericTank tank = TankBuilder.buildTank().at(new Position(1, 1)).with(Attribute.PLAYER_REF, player.toRef()).with(Attribute.DEAD, true).finish();
         State state = TestUtilities.generateBoard(2, 2, tank);
         state.getPlayers().add(player);
+        state.getCouncil().getCouncillors().add(player.toRef());
```

Comment:
Player should not need to be on the council. There might not even be a council in this case. I would like to move to an architecture where the council is not necessary to have a state. There are many of these, please remove them and fix any predicates if necessary.